src/main/java/at/hannibal2/skyhanni/test/command/TrackSoundsCommand.kt

```diff
@@ -0,0 +1,87 @@
+package at.hannibal2.skyhanni.test.command
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.PlaySoundEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.NumberUtil.isInt
+import at.hannibal2.skyhanni.utils.OSUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderables
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.util.concurrent.ConcurrentLinkedDeque
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+object TrackSoundsCommand {
+
+    private var cutOfTime: SimpleTimeMark = SimpleTimeMark.farPast()
+    private var startTime: SimpleTimeMark = SimpleTimeMark.farPast()
+
+    private val sounds = ConcurrentLinkedDeque<Pair<Duration, PlaySoundEvent>>()
+
+    private var isRecording = false
+
+    private val position get() = SkyHanniMod.feature.dev.debug.trackSoundLog
+
+    fun command(args: Array<String>) {
+        if (args.firstOrNull() == "end") {
+            if (!isRecording) {
+                ChatUtils.userError("Nothing to end")
+            } else {
+                cutOfTime = SimpleTimeMark.now()
+            }
+            return
+        }
+        if (isRecording) {
+            ChatUtils.userError("Still tracking sounds, wait for the other tracking to complete before starting a new one or type in \"end\" to end it prematurely")
+            return
+        }
+        isRecording = true
+        sounds.clear()
+        val duration: Duration?
+        if (args.firstOrNull()?.isInt() != true) {
+            duration = null
+            ChatUtils.chat("Now started tracking sounds until manually ended")
+        } else {
+            duration = args.firstOrNull()?.toInt()?.seconds
+            ChatUtils.chat("Now started tracking sounds for ${duration?.inWholeSeconds} Seconds")
+        }
+        startTime = SimpleTimeMark.now()
+        cutOfTime = duration?.let { SimpleTimeMark.future(it) } ?: SimpleTimeMark.farFuture()
```

Comment:
Less code duplication, more "Kotlin like" approach
```suggestion
        cutOfTime = args.firstOrNull()?.toInt()?.seconds?.let {
            ChatUtils.chat("Now started tracking sounds for ${it.inWholeSeconds} Seconds")
            SimpleTimeMark.future(it)
        } ?: run {
            ChatUtils.chat("Now started tracking sounds until manually ended")
            SimpleTimeMark.farFuture()
        }
        startTime = SimpleTimeMark.now()
```